### PR TITLE
ADEN-8562 Fix template slotName value

### PR DESCRIPTION
**Pull request by fraszczakszymon**

### CODE REVIEW

File: src/ad-engine/services/template-service.ts [hunk 1]
```diff
@@ -26,13 +26,20 @@ class TemplateService {
...
 
 		if (!templates[name]) {
 			throw new Error(`Template ${name} does not exist.`);
 		}
 
+		// override params.slotName by splitting it by comma
+		// and picking first occurrence so we can rely on it
+		// in used creative templates
+		if (params && typeof params.slotName === 'string') {
+			params.slotName = params.slotName.split(',').shift();
```
Comment (fraszczakszymon) [inline src/ad-engine/services/template-service.ts]: I had a mixed feeling about overriding incoming params. But on the other hand we've been touched by this issue a few times already and we always forget that slot may have `pos` key-val with many values (array which is passed here as comma separated string). Another approach would be to fix all existing creative templates so they handle slotName:
https://github.com/Wikia/dfp-creative-templates/blob/f785217f0e49b50119d12c7e46ed4693b1f5b64f/hivi-takeover-bfaa-dev.html#L260
like e.g. porvata3 does:
https://github.com/Wikia/dfp-creative-templates/blob/f785217f0e49b50119d12c7e46ed4693b1f5b64f/incontent-porvata-2.html#L51
But then we'll have to remember to do it in all new templates. So probably we'll forget it soon.

Feel free to advise better solution if you see one ;)

File: src/ad-engine/services/template-service.ts [hunk 2]
```diff
@@ -26,13 +26,20 @@ class TemplateService {
...
 		logger(logGroup, 'Load template', name, slot, params);
 
 		if (!templates[name]) {
 			throw new Error(`Template ${name} does not exist.`);
 		}
 
+		// override params.slotName by splitting it by comma
+		// and picking first occurrence so we can rely on it
+		// in used creative templates
+		if (params && typeof params.slotName === 'string') {
```
Comment (jacekbj) [inline src/ad-engine/services/template-service.ts]: if (params && (typeof params.slotName === 'string' || params.slotName instanceof String)) {

File: src/ad-engine/services/template-service.ts [hunk 3]
```diff
@@ -26,13 +26,20 @@ class TemplateService {
 		templates[name] = template;
 	}
 
-	init(name: string, slot: AdSlot = null, params = {}): void {
+	init(name: string, slot: AdSlot = null, params: { [key: string]: any } = {}): void {
```
Comment (jacekbj) [inline src/ad-engine/services/template-service.ts]: You can use new type `Dictionary` from ad-engine/models.

File: src/ad-engine/services/template-service.ts (hunk 1 above)
Comment (nandy-andy) [reply src/ad-engine/services/template-service.ts]: I agree. Cost of keeping such tricky logic here < cost of keeping updating all the templates and keeping it in mind. Do you think we can unit test this part as a documentation of this conversation? ;)

**Pull request merged**